libs/Microsoft.MixedReality.WebRTC/Interop/PeerConnectionInterop.cs

```diff
@@ -814,31 +824,34 @@ internal struct RemoteVideoTrackAddedInfo
         #endregion
 
         #region Utilities
+
         class RemoteDescArgs
         {
-            public ActionDelegate callback;
-            public ManualResetEventSlim completedEvent;
+            public PeerConnectionRemoteDescriptionAppliedDelegate callback;
+            public TaskCompletionSource<bool> tcs;
         }
 
         public static Task SetRemoteDescriptionAsync(PeerConnectionHandle peerHandle, string type, string sdp)
         {
-            return Task.Run(() =>
+            var args = new RemoteDescArgs
+            {
+                callback = RemoteDescriptionApplied,
```

Comment:
The `callback` field is needed to keep the delegate alive, right? Probably needs a comment.